src/Sarif.Multitool/Rules/SARIF1012.MessageArgumentsMustBeConsistentWithRule.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class MessageArgumentsMustBeConsistentWithRule : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF1012
+        /// </summary>
+        public override string Id => RuleId.MessageArgumentsMustBeConsistentWithRule;
+
+        /// <summary>
+        /// Placeholder
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_MessageIdMustExist_Text),
+            nameof(RuleResources.SARIF1012_MessageArgumentsMustBeConsistentWithRule_Error_SupplyEnoughMessageArguments_Text)
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Error;
+
+        private IList<ReportingDescriptor> currentRules;
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            this.currentRules = run?.Tool?.Driver?.Rules;
+        }
+
+        protected override void Analyze(Result result, string resultPointer)
+        {
+            // if message.id is not null, we have to check if that exists in rules
+            if (!string.IsNullOrEmpty(result.Message.Id))
+            {
+                ReportingDescriptor rule = result.RuleIndex >= 0
```

Comment:
>rule [](start = 36, length = 4)

The SARIF SDK has a great helper for this: `result.GetRule(run)`. So:

1. In `Analyze(Run run...)` above, save the `run` in `this.run`.
2. Here, just say

```
ReportingDescriptor rule = result.GetRule(this.run);
```

Reply:
Ok. With that in mind, we will probably change the following condition to this:
```csharp
if (this.currentRules == null
                    || rule.MessageStrings == null
                    || !rule.MessageStrings.ContainsKey(result.Message.Id))
                {
```

Because we dont initialize MessageStrings in the last scenario. What do you think?

---
In reply to: [447332733](https://github.com/microsoft/sarif-sdk/pull/1944#discussion_r447332733) [](ancestors = 447332733)

Reply:
This looks good. You could simplify to
```
if (this.currentRules == null || rule.MessageStrings?.ContainsKey(result.Message.Id) == true)
```
You need the explicit comparison to `true` because the result of a null coalescing operator is a `Nullable<bool>`, not a `bool`, so you can't do an implicit logical test on it.

---
In reply to: [447336153](https://github.com/microsoft/sarif-sdk/pull/1944#discussion_r447336153) [](ancestors = 447336153,447332733)

Reply:
will push soon

---
In reply to: [447339939](https://github.com/microsoft/sarif-sdk/pull/1944#discussion_r447339939) [](ancestors = 447339939,447336153,447332733)